hw-model/src/lib.rs

```diff
@@ -540,6 +540,7 @@ pub trait HwModel {
                 }
             }
             writeln!(hw.output().logger(), "ready_for_fw is high")?;
+            hw.cover_fw_mage(fw_image);
```

Comment:
@rusty1968 to file an issue to address ICCM upgrades. We need to make sure that coverage from the new firmware image doesn't get associated with the old one:

Suggestion:

* Have HwModel::upload_firmware() call cover_fw_image(); this will take care of most of the simple upgrade tests

* For every instruction, compare the instruction read out of iccm with the one read from the covered image, and error if they don't match.